### ISSUE TITLE
HazardStats option: Percentile or SD threshold applied before zonal stat

### ISSUE THREAD

**Issue by matamadio** [MamadioCMCC/HazardRiskScripts#7]
A common issue when producing zonal statistics for aggregating raster data is the presence of outliers. Some times we want to consider them, sometimes we don't, depending on the type of analysis and the type of data. A common solution in this case is to use "majority" or "median" statistics, which sadly most of the times do not deliver as expected.

A good solution was found using the 80th percentile threshold, as calculated on each zone. This approach is more conservative than MAJORITY and more forgiving than MAX.
Example of this two approaches on a landslide vulnerability raster (values 1 to 4):
![immagine](https://user-images.githubusercontent.com/44863827/73008533-0c4c9b00-3e0f-11ea-9960-87c5797cfae8.png)
![immagine](https://user-images.githubusercontent.com/44863827/73008665-50d83680-3e0f-11ea-8034-0dc0dca795d5.png)
![immagine](https://user-images.githubusercontent.com/44863827/73008678-5b92cb80-3e0f-11ea-8f37-61a1999074bf.png)

Another interesting option would be to use standard deviations instead of percentiles threshold, e.g. to set 2 SD from the mean as threshold.

So this is the suggestion to make the script more effective and useful:
- have run option to set any percentile or standard deviation threshold, e.g. PC=80 or SD=2
- the threshold is applied on each zone (e.g. ADM unit) before running the requested statistics. In practice, statistic calculation ignores values found outside the threshold.

**Comment by matamadio** [comment]
Tested it on my pc and server, I got this error:

> python runstats.py --field ADM2_CODE --stats max --pre PC 80 --ncores 3 --indir X:\Work\WB\Thinkhazard\aggregation\test2

```
Traceback (most recent call last):
  File "runstats.py", line 88, in <module>
    filtering = filtering.split(" ")
AttributeError: 'list' object has no attribute 'split'
```

**Comment by ConnectedSystems** [comment]
Should be fixed by ebf042a09ac7496673e4fb8725eb07a91ccbce8f

Let me know how you go

**Comment by ConnectedSystems** [comment]
Closed as this issue is now resolved. Reopen if needed though.
